### ISSUE TITLE
automatically price item based on stash name

### ISSUE THREAD

**Issue by marcusx2** [xyzz/acquisition#118]
I know you can already price multiple items by pricing the stash instead of each item individually. But I noticed that the tool is able to read the stash name and automatically refresh items, so why not create an option where the pricing is based on the stash name? I would suggest using the names defined by poe trade tags. So for example, if a user had a stash called ~b/o 1 chaos, another one called ~c/o 1 chrom, the tool would automatically set the prices for all items on those stash tabs with those prices. I would suggest even setting by default the buyout tag in case the user doesn't specify it, because I think most people use buyout anyway. So for example if the stash name was called just "2 scour" it would actually mean "~b/o 2 scour". You could also allow the user to change the default pricing tag, just in case he/she doesn't use the buyout mostly.

You don't need to remove the option for the way it works right now. You can still let the user set a price for a stash from within the tool, just let it overwrite the price defined from the stash name in the game, or set the price in case the user gave a stash name that the tool can't read(could be a random name like "stuff" or "1" or a name that is wrongly formatted(for example you define that pricing for orb of chance is just called "chance" but the user named the stash "1 orb_chance")). You already do the overwrite thing, since you can price a stash but within the stash you can price items individually.

So the overwrite order would be
1-stash pricing ingame(by using names in the format the tool recognizes)
2-stash pricing in the tool(already exists)
3-individual pricing(already exists)

I think it's a minor enhancement, but very useful. It automates the process even more, basically for anyone who has "specialized"(couldn't come up with a better wording) stash tabs. Users already give pricing names to their stashes in game normally, so they wouldn't need to price the stash twice anymore(in game and in the tool). The less interaction with the tool the better(ironically, but true). 

Thanks for the tool, it's a great program! I look forward to seeing this implemented!

**Comment by Gloorf** [comment]
A way to do that is to connect a slot in buyoutmanager to ItemsRefreshed, and check every tab. If we already have a buyout, we pass. If not, we try to match the tab name to a pattern like you use on poe.trade ("~b/o 1 fuse" for example) ? We can also allows a shortened version by assuming it is a ~b/o[or let user chose default value for this ?] unless explicitely specified in the tab name. Thoughts @xyzz ?

**Comment by xyzz** [comment]
I think we only need to support the ~b/o syntax.

**Comment by Gloorf** [comment]
How about this : https://github.com/Gloorf/acquisition/commit/cb834b140d489bab136ce9f5719616abe488cb7c
If you changes the tab name, it do changes the buyout ; however if you change the buyout from the tab, it.

The big problem is that the buyout based on old tab name isn't removed : for example, if i name my tab "1fuse", there will be a buyout for the tab "1fuse" with a Buyout of ~b/o 1fuse. If i change to "2fuse", the buyout change to ~b/o 2fuse, but there is still a buyout for the tab "1fuse" of ~b/o 1 fuse. How do you manage this problem in acquisition ? I couldn't find it in the code

**Comment by xyzz** [comment]
> How do you manage this problem in acquisition ? I couldn't find it in the code :(

It's not managed in any way. If you set a buyout "1 fuse" for a tab named "1f" it will stay this way. If some other tab gets named "1f" it will also get this buyout set.

**Comment by Gloorf** [comment]
>It's not managed in any way. If you set a buyout "1 fuse" for a tab named "1f" it will stay this way. If some other tab gets named "1f" it will also get this buyout set.

Oh, okay. Then i guess the code is functional then (the string manipulation part can probably be enhanced / changed, i'm pretty bad at those)

**Comment by xyzz** [comment]
Right now it's only going to accept stuff like "1 chaos", "chaos 1", but not "b/o 1 chaos", "price 1 chaos", right? Probably needs more complicated logic.

**Comment by Tvangeste** [comment]
Most players I know use short tab names (so that many tabs fit into the screen without scrolling).

The typical names are 1C, 2c, 5C, 1ex, 2Ex, 5fus, etc.

It would be nice if Acquisition could accept those.

**Comment by Gloorf** [comment]
@xyzz : yes, you're right. I won't be around for the next 10 days or so so don't want to look into this right now, but it shouldn't be too hard to do (then again, with code you never know :P)

@Tvangeste The only problem i see with that is we are going to need a lot of differents abreviations, and people will still complain because their own thing won't be supported :(

**Comment by marcusx2** [comment]
@Gloorf You could do some parsing beforehand. Make it everything lowercase, then remove all spaces. Then you could use some regex(or use regex right from the start) to check the pattern. For example..."[1-9]alch?" would match 1alc and 2alch. I'm pretty sure that with a few regexes it's possible to cover a large amount of possibilities. As Tvangeste said, people usually use short names, so you could make patterns for the first 3 or 4 letters only. I also suggest to support the original names used for poe trade like chaos, scour, and so on.

**Comment by Gloorf** [comment]
i'm back ; have done some improvements (code would probably need a rewrite though) : https://github.com/Gloorf/acquisition/commit/75e72d4a921fcf0066dc33caabca874be21ec8c2

price/no price are now supported, and i've made tentative changes to use different abbreviations and not only poe.trade supported abbreviations.
